core/src/main/scala/slinky/core/annotations/react.scala

```diff
@@ -47,24 +47,30 @@ object ReactMacrosImpl {
         val applyValues = caseClassparamss.map(ps => ps.map(_.name))
 
         val caseClassApply = if (childrenParam.isDefined) {
-          // from https://groups.google.com/forum/#!topic/scala-user/dUOonrP_5K4
-          val body = c.typecheck(childrenParam.get.tpt, c.TYPEmode).tpe match {
-            case TypeRef(_, sym, _) if sym == definitions.RepeatedParamClass =>
-              val applyValuesChildrenVararg = caseClassparamss.map { ps =>
-                ps.map { ps =>
-                  if (ps == childrenParam.get) {
-                    q"${ps.name}: _*"
-                  } else q"${ps.name}"
-                }
+          childrenParam.get.tpt match {
```

Comment:
I wonder if we could make this even more general, based on the comments in #237, by having `expandChildren` make `children` curried regardless of the parameter type. That would line up with JSX behavior better too. And since `expandChildren` is false by default, there's no worry of unexpected behavior.

Reply:
@jedahu thinking about things a bit more, I'm fairly convinced myself that removing the type restriction is the way to go; if you think that makes sense could you go ahead and remove it?